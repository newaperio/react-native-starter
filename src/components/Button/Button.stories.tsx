import { Alert } from "react-native"
import React from "react"
import { storiesOf } from "@storybook/react-native"
import { boolean, text, withKnobs } from "@storybook/addon-knobs"

import Button from "."
import { Centered } from "~storybook/decorators"

const onPress = () => Alert.alert("Boop!")

storiesOf("Button", module)
  .addDecorator(withKnobs)
  .addDecorator(Centered)
  .add("default", () => (
    <Button
      text={text("Text", "Tap Me")}
      onPress={onPress}
      isDisabled={boolean("Disabled?", false)}
    />
  ))
  .add("disabled", () => (
    <Button
      text={text("Text", "Untappable")}
      onPress={onPress}
      isDisabled={boolean("Disabled?", true)}
    />
  ))
