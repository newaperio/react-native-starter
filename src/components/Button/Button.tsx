import React from "react"
import { Text, TouchableOpacity, View } from "react-native"

import { tw } from "~theme"

interface Props {
  isDisabled?: boolean
  onPress: () => void
  text: string
}

export const Button: React.FC<Props> = ({ isDisabled, text, onPress }) => {
  const buttonStyle = "rounded px-8 py-4"

  const ButtonText = () => (
    <Text
      style={tw`font-heading text-center text-${
        isDisabled ? "dark" : "oceanside"
      }`}>
      {text}
    </Text>
  )

  if (isDisabled) {
    return (
      <View style={tw.style(buttonStyle, "bg-stone")}>
        <ButtonText />
      </View>
    )
  } else {
    return (
      <TouchableOpacity
        onPress={isDisabled ? () => null : onPress}
        style={tw.style(buttonStyle, "bg-teal border-2 border-oceanside")}>
        <ButtonText />
      </TouchableOpacity>
    )
  }
}
