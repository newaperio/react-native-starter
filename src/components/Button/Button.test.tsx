import React from "react"
import renderer from "react-test-renderer"

import Button from "."

const onPress = () => null

test("renders correctly", () => {
  const tree = renderer
    .create(<Button text="Snapshot" onPress={onPress} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders as disabled", () => {
  const tree = renderer
    .create(<Button text="Snapshot" onPress={() => null} isDisabled />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
