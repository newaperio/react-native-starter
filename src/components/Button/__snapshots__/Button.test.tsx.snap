// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as disabled 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#999",
      "borderRadius": 4,
      "paddingBottom": 16,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#02181a",
        "fontFamily": "Red Hat Display",
        "textAlign": "center",
      }
    }
  >
    Snapshot
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#40c1bb",
      "borderColor": "#0d5459",
      "borderRadius": 4,
      "borderWidth": 2,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 32,
      "paddingRight": 32,
      "paddingTop": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#0d5459",
        "fontFamily": "Red Hat Display",
        "textAlign": "center",
      }
    }
  >
    Snapshot
  </Text>
</View>
`;
